code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -344,44 +344,92 @@
+	desc = "Channel a sizable vortex of psychic energy, drawing in nearby enemies."
+	ability_name = "Psychic vortex"
+	plasma_cost = 600
+	cooldown_timer = 3 MINUTES
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
 	keybinding_signals = list(
-		KEYBINDING_NORMAL = COMSIG_XENOABILITY_GRAV_NADE,
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_VORTEX,
 	)
-	cooldown_timer = 1 MINUTES
 
-/datum/action/xeno_action/activable/gravity_grenade/use_ability(atom/A)
-	var/turf/T = get_turf(owner)
+/datum/action/xeno_action/activable/psychic_vortex/on_cooldown_finish()
+	to_chat(owner, span_notice("Our mind is ready to unleash another chaotic vortex of energy."))
+	return ..()
+
+/datum/action/xeno_action/activable/psychic_vortex/use_ability(atom/target)
 	succeed_activate()
 	add_cooldown()
-	var/obj/item/explosive/grenade/gravity/nade = new(T)
-	nade.throw_at(A, 5, 1, owner, TRUE)
-	nade.activate(owner)
-
-	owner.visible_message(span_warning("[owner] vomits up a roaring fleshy lump and throws it at [A]!"), span_warning("We vomit up a roaring fleshy lump and throws it at [A]!"))
-
+	addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob, update_icons)), 1 SECONDS)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	owner.icon_state = "[xeno.xeno_caste.caste_name][xeno.is_a_rouny ? " rouny" : ""] Screeching"
+	if(target) // Keybind use doesn't have a target
+		owner.face_atom(target)
+	ADD_TRAIT(owner, TRAIT_IMMOBILE, VORTEX_ABILITY_TRAIT)
+	if(do_after(owner, VORTEX_PULL_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_pull()
+	if(do_after(owner, VORTEX_PUSH_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_push()
+	if(do_after(owner, VORTEX_PULL_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_pull()
+	finish_charging()
+	return fail_activate()
+
+/datum/action/xeno_action/activable/psychic_vortex/proc/vortex_pull()
+	playsound(owner, 'sound/effects/seedling_chargeup.ogg', 60)
+	for(var/atom/movable/victim in range(VORTEX_RANGE, owner.loc))
+		if(victim.anchored)
+			continue
+		if(isliving(victim))
+			var/mob/living_target = victim
+			if(living_target.stat == DEAD)
+				continue
+		victim.throw_at(owner, 3, 1, owner, FALSE)
 
-/obj/item/explosive/grenade/gravity
-	name = "gravity grenade"
-	desc = "A fleshy mass that seems way too heavy for its size. It seems to be vibrating."
-	arm_sound = 'sound/voice/predalien_roar.ogg'
-	greyscale_colors = "#3aaacc"
-	greyscale_config = /datum/greyscale_config/xenogrenade
-	det_time = 20
+	for(var/turf/affected_tile AS in RANGE_TURFS(VORTEX_RANGE, owner.loc))
+		affected_tile.Shake(3, 3, 1 SECONDS)
+		for(var/atom/movable/affected AS in affected_tile)
+			if(ishuman(affected))
+				var/mob/living/carbon/human/H = affected
+				if(H.stat == DEAD)
+					continue
+				H.apply_effects(1,1)
+				H.adjust_stagger(2)
+				shake_camera(H, 2, 1)
+				H.Shake(3, 3, 10)
+				affected.Shake(3, 3, 10)
+				continue
+			if(isitem(affected) && isdroid(affected))
+				affected.Shake(3, 3, 10)
+				continue
```

Comment:
This is super wack with the 3 for loops.
Do a single loop and check for everything there.